src/Engine/RulesComponent.php

```diff
@@ -228,13 +228,23 @@ public function executeWithArguments(array $arguments) {
    *   A list object containing \Drupal\rules\Engine\IntegrityViolation objects.
    */
   public function checkIntegrity() {
+    $metadata_state = $this->getMetadataState();
+    return $this->expression->checkIntegrity($metadata_state);
+  }
+
+  /**
+   * Gets the metadata state.
+   *
+   * @return \Drupal\rules\Engine\ExecutionMetadataStateInterface
+   *   The populated execution metadata state.
```

Comment:
Should be clarified that it's the state at component invocation time imo. it does not describe the state after execution, but before execution.